asf_heat_pump_suitability/utils/geo_utils.py

```diff
@@ -88,3 +89,82 @@ def ensure_crs_match(
     if gdf_1.crs != gdf_2.crs:
         gdf_2 = gdf_2.to_crs(gdf_1.crs)
     return gdf_2
+
+
+def load_and_filter_lsoa_geometries(
+    la_lsoas: List[str], lsoa_shp_path: str, target_crs: str
+) -> gpd.GeoDataFrame:
+    """
+    Loads and preprocesses LSOA geospatial data by:
+      - Reading a shapefile.
+      - Converting to a target CRS if necessary.
+      - Filtering LSOAs to just those that belong to the specified local authority.
+
+    Args:
+        la_lsoas (List[str]): List of LSOA codes for the current local authority.
+        lsoa_shp_path (str): Path (local or S3) to the national LSOA shapefile.
+        target_crs (str): The target Coordinate Reference System (e.g., "EPSG:27700").
+
+    Returns:
+        gpd.GeoDataFrame: GeoDataFrame with filtered LSOA geometries in the target CRS.
+
+    Raises:
+        ValueError: If no matching LSOAs are found in the shapefile.
```

Comment:
```suggestion
    lsoas: List[str], lsoa_shp_path: str, target_crs: str
) -> gpd.GeoDataFrame:
    """
    Loads and preprocesses LSOA geospatial data by:
      - Reading a shapefile.
      - Converting to a target CRS if necessary.
      - Filtering LSOAs to just those specified.

    Args:
        lsoas (List[str]): List of LSOA codes to get geometries for.
        lsoa_shp_path (str): Path (local or S3) to the national LSOA shapefile.
        target_crs (str): The target Coordinate Reference System (e.g., "EPSG:27700").

    Returns:
        gpd.GeoDataFrame: GeoDataFrame with filtered LSOA geometries in the target CRS.

    Raises:
        ValueError: If no matching LSOAs are found in the shapefile.
```
This can be made more generic :)